src/db.c

```diff
@@ -880,6 +888,46 @@ int parseScanCursorOrReply(client *c, robj *o, unsigned long *cursor) {
     return C_OK;
 }
 
+char *obj_type_name[OBJ_TYPE_MAX] = {
+    "string", 
+    "list", 
+    "set", 
+    "zset", 
+    "hash", 
+    "", /* module type is special */
+    "stream"
+};
+
+long long getObjectTypeByName(char *name) {
+    for (long long i = 0; i < OBJ_TYPE_MAX; i++) {
+        if (!strcasecmp(name, obj_type_name[i])) {
+            return i;
+        }
+    }
+
+    moduleType *mt = moduleTypeLookupModuleByName(name);
+    if (mt != NULL) return -(REDISMODULE_TYPE_SIGN(mt->id));
```

Comment:
are we sure this will never return values in the range of 0 to 7, or LLONG_MAX?
that's a 54 bit number, so negating it seems safe, just making sure..

Reply:
yes, although long long type size is platform-dependent, but guaranteed by the C standard (ISO C99) to be at least 64 bits, so we don't need to worry about it